jerry-core/api/jerry.c

```diff
@@ -3458,14 +3458,19 @@ jerry_create_arraybuffer_external (const jerry_length_t size, /**< size of the b
   jerry_assert_api_available ();
 
 #if ENABLED (JERRY_ES2015_BUILTIN_TYPEDARRAY)
-  if (size == 0 || buffer_p == NULL)
+  ecma_object_t *arraybuffer;
+
+  if (JERRY_UNLIKELY (size == 0 || buffer_p == NULL))
+  {
+    arraybuffer = ecma_arraybuffer_new_object_external (0, NULL, free_cb);
+  }
+  else
   {
-    return jerry_throw (ecma_raise_range_error (ECMA_ERR_MSG ("invalid buffer size or storage reference")));
+    arraybuffer = ecma_arraybuffer_new_object_external (size,
+                                                        buffer_p,
+                                                        (ecma_object_native_free_callback_t) free_cb);
```

Comment:
Why this casting is needed? I know it is in the old code, still :)

Reply:
The type of the cb parameter is `jerry_object_native_free_callback_t` while `ecma_arraybuffer_new_object_external` expects `ecma_object_native_free_callback_t` typed cb. Both type is `void*`. Would you like to remove this casting?

Reply:
A line above you wrote:
`arraybuffer = ecma_arraybuffer_new_object_external (0, NULL, free_cb);`
Based on this, the cast is unneeded.

Reply:
I see. I've added missing conversion to the `arraybuffer = ecma_arraybuffer_new_object_external (0, NULL, free_cb);` line.